### DISCUSSION TITLE
The behavior of `throw createError` in `/server/api/x.ts` and `/server/routes/x.ts` is different.

### DISCUSSION THREAD

**Discussion by mkdirnow** [nuxt/nuxt#28167] [Questions] [answered]
The code in the two `x.ts` files is identical: 
```
export default eventHandler(() => {
  throw createError({
    statusCode: 401,
  });
});
```

However, they have different HTTP responses: 

![image](https://github.com/user-attachments/assets/7c494c89-ec7d-46d8-96d2-9cdb94000b5f)

How to return the same error format in `/server/routes` as in `/server/api`?

**Comment by masyoudi** [comment, marked as answer]
i think the error handler for `/server/api` is returning json and `/server/routes` is returning html, you can check the error handler from the [source code](https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/core/runtime/nitro/error.ts)

or you can customize the error handler with nuxt.config
```ts
export default defineNuxtConfig({
  nitro: {
    errorHandler: 'path-to-your-custom-error-handler.ts'
  }
})
```

**Reply by mkdirnow** [reply]
Thank you for the explanation. I never noticed Nitro's `errorHandler`. Thanks for the idea.
